### DISCUSSION TITLE
优化导出错误报告的内容

### DISCUSSION THREAD

**Discussion by whitecat346** [Meloong-Git/PCL#5194] [讨论]
## 前言
错误报告作为当游戏出现错误的时候唯一一个他人可以帮助判断错误原因的途径，应当包含尽量详细的信息。但是对于目前PCL2错误报告的信息实在不是很详细，所以这里就来讨论一下增加错误报告信息的提议。

## 当前版本的弊端
目前的错误报告为一个名称为“错误报告-年份-月份-日期_时.分.秒”的ZIP压缩文件，包含：
- 用于查询版本信息的清单文件(.json)
- 最后的日志(latest.log)
- PCL 启动器日志(.txt）
- 启动脚本(.bat)
- 游戏崩溃前的输出(.txt)

虽然用于普通错误的查询已经足够的，但是在一些情况下（如：用户名非法、Java版本相关的异常、系统环境错误等）依然不足以用于分析错误。

### 1、可以随时导出错误报告
在游戏启动后可能会遇到游戏长时间未响应的问题，这个时候可以选择导出当前游戏日志，用于评判游戏是否处于卡死的情况。当然，可以在游戏运行的时候进行导出。
### 2、增加错误报告压缩包的内容
当前错误报告压缩包内容有点有限，应当增加一个包含了：
- 系统版本
- 硬件信息
- 内存和存储信息
- 游戏版本、MOD加载器（如果有）、运行游戏的Java信息、JVM参数
- 一些视情况而定的MOD、资源包、光影包列表
- 玩家信息（不可包含隐私信息）
- 待定……
等信息的文件（不限制为单文件，可以多个）。这样可以增加分析日志时的便捷。
### 3、对于PCL2自身的错误增加导出错误报告的途径
一些PCL2自身出现的报错没有导出错误报告的途径，无法便捷的反馈错误，对于分析出错原因十分不便，故在这里建议增加一个导出PCL2自身错误报告的途径。

## 后记
这个提议只是一时想出的，一些细节并未经过推敲，还请看到的各位提出一些建议作为补充。

**Comment by Pigeon0v0** [comment]
很多信息其实启动器日志里都有，启动器本体实际上也只需要日志就足以判断问题...

**Reply by whitecat346** [reply]
我提这个主要是为了能够给一个将这些信息汇总到一块的地方，启动器日志里的一些无关信息太多，锁定目标太麻烦，而且不是log格式文件，用vscode之类的打开没有高亮🧐

**Reply by 3gf8jv4dv** [reply]
> 而且不是log格式文件，用vscode之类的打开没有高亮🧐

VSCode 界面右下角「选择语言模式」选择「Log」就有高亮了。

**Reply by MoYuan-CN** [reply]
我倒是配置了 `regexp Log*.txt` 自动解析为 `Log` 语言

**Reply by whitecat346** [reply]
> > 而且不是log格式文件，用vscode之类的打开没有高亮🧐
> 
> VSCode 界面右下角「选择语言模式」选择「Log」就有高亮了。

经常忘了这个……

**Comment by whitecat346** [comment]
~~补充：建议对离线模式下对用户名命名Unicode字符时添加警告，因为这可能导致游戏无法正常游玩~~

**Reply by hejiehao** [reply]
哪些 Unicode 字符？

**Reply by whitecat346** [reply]
中文之类的那些mc原版不支持的

**Reply by hejiehao** [reply]
![image](https://github.com/user-attachments/assets/a0988148-5651-42fd-a45f-ac887456cfac)

**Reply by whitecat346** [reply]
看来是已经有了……那就忽视这个提议吧
